docs/extensions/extension-registry.md

```diff
@@ -0,0 +1,580 @@
+
+To understand what information is needed to _model and register your own instance of AI/ML tool or service as a FuseML extension_ and then _reference it and access it from FuseML workflows_, please proceed to the next sections describing the [Extension Record format](#extension-records) and the [workflow extension references](#referencing-extensions-in-workflows).
+
+To explore more on the use-cases for which the Extension Registry was intended, please refer to the [Extension Registry Use Cases section](#extension-registry-use-cases).
+
+## Extension Records
+
+The structure of the information captured in the Extension Registry (i.e. the Extension Records) is based on common architectural and deployment patterns and abstractions exhibited by and extracted from a wide range of AI/ML tools and services, but is particularly suited to describe cloud-native applications and services. The Extension Record is represented by a hierarchy of elements (services, endpoints, credentials and configuration) that can be used to model any 3rd party service in a form that can be easily referenced and consumed by FuseML and automated workflows.
+
+Below is an example of an Extension Record describing an MLFlow server instance deployed in a Kubernetes cluster. In fact, it is the exact Extension Record that is automatically registered when MLFlow is installed using the FuseML installer.
+
+```yaml
+id: mlflow
+product: mlflow
+version: 1.20.2
+description: |
+  MLFlow is an open source platform specialized in tracking ML experiments, and packaging and deploying ML models.
+services:
+- id: mlflow-tracking
+  resource: mlflow-tracking
+  category: experiment-tracking
+  authrequired: false
+  description: MLFlow experiment tracking service API and UI
+  endpoints:
+  - url: http://mlflow.mlflow
+    type: internal
+    configuration:
+      MLFLOW_TRACKING_URI: http://mlflow.mlflow
+- id: mlflow-store
+  resource: s3
+  category: model-store
+  authrequired: true
+  description: MLFlow minio S3 storage back-end
+  endpoints:
+  - url: http://mlflow-minio.mlflow:9000
+    type: internal
+    configuration:
+      MLFLOW_S3_ENDPOINT_URL: http://mlflow-minio.mlflow:9000
+  credentials:
+  - id: default-s3-account
+    default: false
+    scope: global
+    projects: []
+    users: []
+    configuration:
+      AWS_ACCESS_KEY_ID: <hidden>
+      AWS_SECRET_ACCESS_KEY: <hidden>
+```
+
+As can be seen from the descriptor, an MLFlow server consists of two services: the experiment tracking service and the artifact store. The experiment tracking service is responsible for tracking and visualizing ML experiments. The artifact store (in our case backed by minio - an S3 object store) is the service responsible for storing ML artifacts, such as ML models, datasets and logs. Services have individual endpoints through which they can be accessed. The artifact store in particular requires authentication, which is why credentials are also provided.
```

Comment:
Maybe add a hyperlink for minio page here... ?

Reply:
done